src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -410,6 +414,11 @@ public function access($op, DataInterpreterInterface $interpreter) {
    * @throws ServerConfigurationException
    */
   protected function propertyWrapper(DataInterpreterInterface $interpreter) {
+    // This is the first method that gets called for all fields after loading
+    // the entity. We'll use that opportunity to set the actual bundle of the
+    // field.
+    $this->setBundle($interpreter->getWrapper()->getBundle());
```

Comment:
This gets constructed with the `Resource` when the request comes in. That means that the entity is not loaded yet, which in turn means that we don't know the bundle of the entity.

After loading the entity this is the first method that gets called for all entity-based resource fields that implies an entity being loaded and has access to it through the `$interpreter`.